Feature/project crud

muhammad-back opened the pull request.

muhammad-back committed:
feat(API): added controllers routes to projecttypes

muhammad-back committed:
feat: added new foutes for worktimetracking and projects

abubakrmirgiyasov commented on src/WorkTimeTracker.API/Models/Requests/UpdateWorkTimeTrackingRequest.cs:
```diff
@@ -0,0 +1,19 @@
+public class UpdateWorkTimeTrackingRequest
+{
+
+    public required string Name { get; set; }
```
`set -> init`

abubakrmirgiyasov commented on src/WorkTimeTracker.API/Models/Requests/UpdateWorkTimeTrackingRequest.cs:
```diff
@@ -0,0 +1,19 @@
+
+    public required string Name { get; set; }
+
+    public string? Description { get; set; }
```
`set -> init`

abubakrmirgiyasov commented on src/WorkTimeTracker.API/Models/Requests/UpdateWorkTimeTrackingRequest.cs:
```diff
@@ -0,0 +1,19 @@
+
+    public string? Description { get; set; }
+
+    public required string ShortDescription { get; set; }
```
`set -> init`

abubakrmirgiyasov commented on src/WorkTimeTracker.API/Models/Requests/UpdateWorkTimeTrackingRequest.cs:
```diff
@@ -0,0 +1,19 @@
+
+    public required string ShortDescription { get; set; }
+
+    public required DateTime DateMin { get; set; }
```
`set -> init`

abubakrmirgiyasov commented on src/WorkTimeTracker.API/Models/Requests/UpdateWorkTimeTrackingRequest.cs:
```diff
@@ -0,0 +1,19 @@
+
+    public required DateTime DateMin { get; set; }
+
+    public required DateTime DateMax { get; set; }
```
`set -> init`

abubakrmirgiyasov commented on src/WorkTimeTracker.API/Models/Requests/UpdateWorkTimeTrackingRequest.cs:
```diff
@@ -0,0 +1,19 @@
+namespace WorkTimeTracker.API.Models.Requests;
+
+public class UpdateWorkTimeTrackingRequest
```
`public sealed class UpdateWorkTimeTrackingRequest`

abubakrmirgiyasov commented on src/WorkTimeTracker.API/Models/Requests/UpdateWorkTimeTrackingRequest.cs:
```diff
@@ -0,0 +1,19 @@
+
+    public required DateTime DateMax { get; set; }
+
+    public required DateTime Time { get; set; }
```
`set -> init`

abubakrmirgiyasov commented on src/WorkTimeTracker.API/Models/Requests/UpdateWorkTimeTrackingRequest.cs:
```diff
@@ -0,0 +1,19 @@
+
+    public required DateTime Time { get; set; }
+
+    public required long ProjectTypeId { get; set; }
```
`set -> init`

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/Projects/Commands/UpdateProjectCommand.cs:
```diff
@@ -0,0 +1,70 @@
+public sealed class UpdateProjectCommandValidator : AbstractValidator<UpdateProjectCommand>
+{
+    public UpdateProjectCommandValidator()
+    {
```
Нету проверки на Id

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/Projects/ProjectErrors.cs:
```diff
@@ -0,0 +1,11 @@
+{
+    public static readonly Error NotFound = new(
+        "Project.NotFound",
+        "Тип проекта не найден."
```
Это не тип проекта

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/WorkTimeTrackings/Commands/UpdateWorkTimeTrackingCommand.cs:
```diff
@@ -0,0 +1,82 @@
+            workTimeTracking.SetDescription(request.Description);
+
+        workTimeTracking.SetName(request.Name);
+
```
удалить лишний Enter

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/WorkTimeTrackings/Commands/UpdateWorkTimeTrackingCommand.cs:
```diff
@@ -0,0 +1,82 @@
+        workTimeTracking.SetName(request.Name);
+
+        workTimeTracking.SetShortDescription(request.ShortDescription);
+
```
удалить лишний Enter

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/WorkTimeTrackings/Commands/UpdateWorkTimeTrackingCommand.cs:
```diff
@@ -0,0 +1,82 @@
+        workTimeTracking.SetShortDescription(request.ShortDescription);
+
+        workTimeTracking.SetDateMin(request.DateMin);
+
```
удалить лишний Enter

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/WorkTimeTrackings/Commands/UpdateWorkTimeTrackingCommand.cs:
```diff
@@ -0,0 +1,82 @@
+        workTimeTracking.SetDateMin(request.DateMin);
+
+        workTimeTracking.SetDateMax(request.DateMax);
+
```
удалить лишний Enter

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/WorkTimeTrackings/Commands/UpdateWorkTimeTrackingCommand.cs:
```diff
@@ -0,0 +1,82 @@
+        workTimeTracking.SetDateMax(request.DateMax);
+
+        workTimeTracking.SetTime(request.Time);
+
```
удалить лишний Enter

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/WorkTimeTrackings/Queries/GetWorkTimeTrackingById.cs:
```diff
@@ -0,0 +1,42 @@
+
+namespace WorkTimeTracker.Application.WorkTimeTrackings.Queries;
+
+public sealed record GetWorkTimeTrackingById(long id) : IRequest<WorkTimeTrackingDto>;
```
`id -> Id`

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/WorkTimeTrackings/WorkTimeTrackingErrors.cs:
```diff
@@ -0,0 +1,10 @@
+{
+    public static readonly Error NotFound = new(
+        "WorkTimeTracking.NotFound",
+        "Тип проекта не найден."
```
Причем тут тип проекта???

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/Projects/Commands/CreateProjectCommand.cs:
```diff
@@ -0,0 +1,64 @@
+    public CreateProjectCommandValidator()
+    {
+        RuleFor(x => x.Link).NotEmpty().When(x => !string.IsNullOrWhiteSpace(x.Link));
+
```
удалить лишний Enter

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/Projects/Commands/CreateProjectCommand.cs:
```diff
@@ -0,0 +1,64 @@
+        RuleFor(x => x.Link).NotEmpty().When(x => !string.IsNullOrWhiteSpace(x.Link));
+
+        RuleFor(x => x.Name).NotEmpty();
+
```
удалить лишний Enter

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/Projects/Commands/DeleteProjectCommand.cs:
```diff
@@ -0,0 +1,40 @@
+        RuleFor(x => x.Id).GreaterThan(0);
+    }
+}
+internal sealed class DeleteProjectCommandHandler : IRequestHandler<DeleteProjectCommand>
```
добавить Enter

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/Projects/Commands/UpdateProjectCommand.cs:
```diff
@@ -0,0 +1,70 @@
+        if (!string.IsNullOrEmpty(request.Link))
+            project.SetLink(request.Link);
+
+        project.SetName(request.Name);
```
перенести вверх в 55строку

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/Projects/Queries/GetProjectById.cs:
```diff
@@ -0,0 +1,42 @@
+
+namespace WorkTimeTracker.Application.Projects.Queries;
+
+public sealed record GetProjectById(long id) : IRequest<ProjectDto>;
```
`id -> Id`

muhammad-back committed:
fix(IO): fixed updateworktimetrackingrequest fields from set to init

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/Projects/Commands/CreateProjectCommand.cs:
```diff
@@ -0,0 +1,67 @@
+            .NotEmpty()
+            .When(x => !string.IsNullOrWhiteSpace(x.Link));
+        RuleFor(x => x.Name)
+            .NotEmpty();
```
сделать в ряд

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/Projects/Commands/DeleteProjectCommand.cs:
```diff
@@ -0,0 +1,41 @@
+    public DeleteProjectCommandValidator()
+    {
+        RuleFor(x => x.Id)
+            .GreaterThan(0);
```
сделать в ряд

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/Projects/Commands/DeleteProjectCommand.cs:
```diff
@@ -0,0 +1,41 @@
+            .GreaterThan(0);
+    }
+}
+internal sealed class DeleteProjectCommandHandler : IRequestHandler<DeleteProjectCommand>
```
Добавить Enter

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/Projects/Commands/UpdateProjectCommand.cs:
```diff
@@ -0,0 +1,67 @@
+    public UpdateProjectCommandValidator()
+    {
+        RuleFor(x => x.Id)
+            .GreaterThan(0);
```
сделать в ряд

abubakrmirgiyasov commented on src/WorkTimeTracker.Application/Projects/Commands/UpdateProjectCommand.cs:
```diff
@@ -0,0 +1,67 @@
+
+        if (project is null)
+            throw new ResourceNotFoundException(ProjectErrors.NotFound);
+        if (!string.IsNullOrEmpty(request.Link))
```
Добавить Enter

muhammad-back committed:
fixed projects commands

muhammad-back committed:
fix(IO): fixed updateworktimetrackingrequest fields from set to init
fixed projects commands
fixed projects queries getprojectbyid fields from id to Id
fixed projecterrors
fixed worktimetracker commands updateworktimetrackingcommand
fixed worktimetracker queries getworktimetrackingbuid
fixed worktimetrackingerrors
